datafusion/common/src/dfschema.rs

```diff
@@ -347,7 +347,20 @@ impl DFSchema {
         matches.next()
     }
 
-    /// Find the index of the column with the given qualifier and name
+    /// Find the index of the column with the given qualifier and name,
+    /// returning `None` if not found
+    ///
+    /// See [Self::index_of_column] for a version that returns an error if the
+    /// column is not found
+    pub fn maybe_index_of_column(&self, col: &Column) -> Option<usize> {
+        self.index_of_column_by_name(col.relation.as_ref(), &col.name)
```

Comment:
This hung me up for a while -- the original code uses `flat_map` which discards the `Err` if the column is not found, was not obvious to me

Reply:
This is much more clear, Thanks!